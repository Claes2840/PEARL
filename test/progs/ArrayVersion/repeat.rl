init: entry
      if n = 0 goto stop else loop

loop: fi empty s from init else loop
      push x s
      x += top s
      n -= 1
      if n = 0 goto stop else loop

stop: fi empty s from init else loop
      exit