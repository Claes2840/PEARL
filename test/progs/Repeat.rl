// repeat x n = list of n+1 x's

init: entry
      if n = '0 goto stop else loop

loop: fi list = 'nil from init else loop
      list <- (x . list)
      x ^= hd list
      n -= '1
      if n = '0 goto stop else loop

stop: fi list = 'nil from init else loop
      list <- (x . list)
      exit