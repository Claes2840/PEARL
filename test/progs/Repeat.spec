list = 'nil
n = '3