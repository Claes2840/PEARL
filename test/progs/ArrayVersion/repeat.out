init_n_3:
	entry
	goto loop_n_3


loop_n_3:
	fi empty s
		from init_n_3
		else loop_NULL
	push x s
	x += top s
	goto loop_n_2


loop_n_2:
	fi empty s
		from init_NULL
		else loop_n_3
	push x s
	x += top s
	goto loop_n_1


loop_n_1:
	fi empty s
		from init_NULL
		else loop_n_2
	push x s
	x += top s
	goto stop_n_0


stop_n_0:
	fi empty s
		from init_NULL
		else loop_n_1
	exit